lib/shipit/line_buffer.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module Shipit
+  class LineBuffer
+    SEPARATOR = "\n"
+
+    def initialize(value = "")
+      @value = value
+    end
+
+    def buffer(text, &block)
+      @value += text
+      whole_lines.each(&block).tap { flush }
```

Comment:
Why the `.tap { }` ? Seems like flushing on the next line would be just fine.

Reply:
In order to return the Enumerator.